src/controller/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+project(NWNx4Controller VERSION 1.0)
+
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /DWIN32 /D_DEBUG /D_CONSOLE /D_CRT_SECURE_NO_DEPRECATE /D__WXMSWUD__ /D__WXDEBUG__ /D_WINDOWS")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /DWIN32 /DNDEBUG /D_CONSOLE /D_CRT_SECURE_NO_DEPRECATE /D__WXMSWU__ /D_WINDOWS")
+set(SOURCE_FILES ../misc/log.cpp ../misc/log.h main.cpp controller.cpp service.cpp udp.cpp)
+
+add_executable(NWNx4Controller ${SOURCE_FILES})
+find_package(gRPC CONFIG REQUIRED)
+
+if(${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+    target_include_directories(NWNx4Controller PRIVATE ../controller ../../lib/NWN2/include ${VCPKG_INCLUDE_DIR} ${VCPKG_LIB_DIR_DEBUG}/mswud)
+    target_link_directories(NWNx4Controller PRIVATE ../lib/NWN2 ${VCPKG_LIB_DIR_DEBUG})
+    target_link_libraries(NWNx4Controller PRIVATE ws2_32.lib wsock32.lib detours.lib wxmsw31ud_core.lib wxbase31ud.lib wxregexud.lib gRPC::grpc++)
+elseif(${CMAKE_BUILD_TYPE} STREQUAL "Release")
+    target_include_directories(NWNx4Controller PRIVATE ../controller ../../lib/NWN2/include ${VCPKG_INCLUDE_DIR} ${VCPKG_LIB_DIR_RELEASE}/mswu)
+    target_link_directories(NWNx4Controller PRIVATE ../lib/NWN2 ${VCPKG_LIB_DIR_RELEASE})
+    target_link_libraries(NWNx4Controller PRIVATE ws2_32.lib wsock32.lib detours.lib wxmsw31u_core.lib wxbase31u.lib wxregexu.lib gRPC::grpc++)
```

Comment:
It seems architecturally unfortunate to include gRPC in the NWNX controller.  The controller's job is purely to orchestrate starting/stopping nwn2server, monitor its responsiveness and deal with recovery from crashes, and to bootstrap the process of loading plugins into nwn2server.

gRPC seems to have a non-trivial network-accessible attack surface, and so it's reasonably likely that anything linking to gRPC will need to be updated (and all servers will have to take updates etc.) when there are inevitable security vulnerabilities found & fixed upstream in gRPC.  Ideally, it seems best to only burden environments that are going to use the gRPC-based functionality with the need to rapidly track & deploy any such fixes... can any future logic to use gRPC live, instead, in its own plugin, so that only environments that deploy that plugin need to worry about any possible gRPC issues (especially if there's listener-like code that would be involved)?

Reply:
This was boilerplate code to get the work running first, then isolate it and place it into another plugin. It is being moved into an xp_rpc plugin that I'm building on my end.